server/src/main/java/org/candlepin/controller/CandlepinPoolManager.java

```diff
@@ -343,7 +343,7 @@ void refreshPoolsForMasterPool(Pool pool, boolean updateStackDerived, boolean la
 
         // These don't all necessarily belong to this owner
         List<Pool> subscriptionPools;
-
+        pool.setLocked(true);
```

Comment:
Try not to interlace logical blocks operations. This is placed right in the middle of the declaration for the sub pools and the code which populates it with no whitespace to separate it visually.

Reply:
yes sure.